src/NUnitEngine/nunit.engine/Services/DomainManager.cs

```diff
@@ -123,11 +118,19 @@ public AppDomainSetup CreateAppDomainSetup(TestPackage package)
                 appBase += Path.DirectorySeparatorChar;
 
             setup.ApplicationBase = appBase;
-            // TODO: Check whether Mono still needs full path to config file...
-            setup.ConfigurationFile = appBase != null && configFile != null
-                ? Path.Combine(appBase, configFile)
-                : configFile;
 
+            // TODO: Check whether Mono still needs full path to config file...
+            string configurationFile;
+            if (TryGetConfigFileName(
```

Comment:
The old code always set the setup.ConfigurationFile, even if it was set to an empty string. In that case, wouldn't this code read better as,

```C#
setup.ConfigurationFile = GetConfigFileName(
                package.GetSetting(PackageSettings.ConfigurationFile, string.Empty),
                testFile != null ? testFile.FullName : string.Empty,
                appBase);
```

The GetConfigFileName would just return string.Empty for the one false case in the method. I might then refactor to

```C#
var settingsConfigFile = package.GetSetting(PackageSettings.ConfigurationFile, string.Empty);
var testFileName =  testFile != null ? testFile.FullName : string.Empty;
setup.ConfigurationFile = GetConfigFileName(settingsConfigFile, testFileName,  appBase);
```

There is just a bit too much going on in this if statement to make it readable.